test/jdk/javax/swing/JTableHeader/TableRendererTest.java

```diff
@@ -0,0 +1,131 @@
+
+public class TableRendererTest{
+    static Boolean PassFailStatus = true;
+    static JFrame frame = null;
+    private static JScrollPane jScrollPane = null;
+    private static JTable table = null;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("com.sun.java.swing.plaf.windows.WindowsLookAndFeel");
+
+        Robot robot = new Robot();
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                try {
+                    Thread.currentThread().setUncaughtExceptionHandler(new ExceptionCheck());
+                    initialize();
+                    //Adding the Test Frame to handle dispose
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+
+        robot.delay(2000);
+        robot.waitForIdle();
+
+        if (!PassFailStatus) {
+            frame.dispose();
+            throw new RuntimeException("Test Case Failed. NPE raised!");
+        }
+        System.out.println("Test Pass!");
+        frame.dispose();
+    }
+
+    static void initialize() {
+        frame = new JFrame();
+        frame.setTitle("JFrame");
+        table = new JTable(new MyModel());
+        jScrollPane = new JScrollPane();
+        jScrollPane.setBounds(new java.awt.Rectangle(0,0,350,618));
+        jScrollPane.setViewportView(table);
+
+        JTableHeader header = table.getTableHeader();
+        TableCellRenderer render = new DecoratedHeaderRenderer(header.getDefaultRenderer());
+        header.setDefaultRenderer(render);
+
+        frame.getContentPane().add(jScrollPane,null);
+        frame.setSize(363, 658);
+        frame.setDefaultCloseOperation(javax.swing.WindowConstants.DISPOSE_ON_CLOSE);
+
+        table.updateUI();
+
+        frame.setVisible(true);
+    }
+
+    static class ExceptionCheck implements Thread.UncaughtExceptionHandler {
+        public void uncaughtException(Thread t, Throwable e)
+        {
+            PassFailStatus = false;
+            System.out.println("Uncaught Exception Handled : " + e);
+        }
+    }
+
+    static class MyModel extends AbstractTableModel {
+        public int getRowCount() {
+            return 10;
+        }
+        public int getColumnCount() {
+            return 10;
+        }
+        public Object getValueAt(int rowIndex, int columnIndex) {
+            return ""+rowIndex + " X " + columnIndex;
+        }
+    }
+
+    static class DecoratedHeaderRenderer implements TableCellRenderer {
+        public DecoratedHeaderRenderer(TableCellRenderer render) {
+            this.render = render;
+        }
+
+        private final TableCellRenderer render;
+
+        public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
+            return render.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, column);
+        }
```

Comment:
I suggest breaking these long lines. You can list parameters as in the interface declaration:
https://github.com/openjdk/jdk/blob/3789983/src/java.desktop/share/classes/javax/swing/table/TableCellRenderer.java#L94-L96